app/scripts/views/mixins/signed-out-notification-mixin.js

```diff
@@ -34,11 +33,9 @@ define(function (require, exports, module) {
     },
 
     navigateToSignIn () {
-      this.navigate('signin', {
-        success: BaseView.t('Signed out successfully')
-      }, {
-        clearQueryParams: true
-      });
+      // Navigate via the back-end in order to regenerate
+      // flow id and flow begin time for the next session.
+      this.window.location.href = '/signin';
```

Comment:
👍 - I was just looking at how I do a similar thing in [views/connect_another_device.js](https://github.com/mozilla/fxa-content-server/blob/6fa4336c19b255fff4657045a12b85eb9f8270b1/app/scripts/views/connect_another_device.js#L226:L247) and realized a few things.

First, there should probably be an abstraction for the action of navigating via location.href instead via pushState. The reason is we want to ensure all metrics are flushed before redirecting and not all environments support the beforeunload handler, Safari on iOS is one example. With an abstraction, we could do something like:

```js
navigateToUrl (url) {
   return this.metrics.flush()
     .then(() => this.window.location.href = url);
},
```

We have the this concept in [external-links-mixin.js->_flushMetricsThenRedirect](https://github.com/mozilla/fxa-content-server/blob/6fa4336c19b255fff4657045a12b85eb9f8270b1/app/scripts/views/mixins/external-links-mixin.js#L126), maybe that could be surfaced as a public method, or maybe we could move that logic to [router.js->navigate](https://github.com/mozilla/fxa-content-server/blob/6fa4336c19b255fff4657045a12b85eb9f8270b1/app/scripts/lib/router.js#L141) with a new option `updateLocation: true` or similar and have the external-links-mixin use that?

The second thing I noticed is that the connect another device page doesn't handle this either for the sign in button. Fortunately the sign in button is not displayed in Safari for iOS, so we should be good. You don't need to fix this, it's a note for me to open an issue about it.

The third thing I noticed is that we should probably propagate the current query parameters. If a user disconnects their current device when accessing /settings from within "Manage Account", then propagating the query params should allow the user to sign in again.